extras/reps_and_likelihoods.cpp

```diff
@@ -0,0 +1,82 @@
+#include "reps_and_likelihoods.hpp"
+
+int main(int argc, char *argv[]) {
+  // std::cout << argc << std::endl;
+  if (argc != 8) {
+    std::cout
+        << "We need exactly 7 arguments: fasta, rooted_nwk, credible_rooted_nwk,"
+           "pp_rooted_nwk, repr_out_path, credible_repr_out_path, and pp_repr_out_path"
+        << std::endl;
+    abort();
+  }
+  std::string fasta_path = argv[1];
+  std::string rooted_nwk_path = argv[2];
+  std::string credible_rooted_nwk_path = argv[3];
+  std::string pp_rooted_nwk_path = argv[4];
+  std::string out_path = argv[5];
+  std::string credible_out_path = argv[6];
+  std::string pp_out_path = argv[7];
+  auto thread_count = std::thread::hardware_concurrency();
+
+  GPInstance all_trees_gp_inst("mmapped_plv.data");
+  RootedSBNInstance cred_r_inst("cred_trees");
+  RootedSBNInstance pp_r_inst("pp_trees");
+  all_trees_gp_inst.ReadNewickFile(rooted_nwk_path);
+  all_trees_gp_inst.ReadFastaFile(fasta_path);
+  all_trees_gp_inst.MakeEngine();
+  all_trees_gp_inst.TakeFirstBranchLength();
+  cred_r_inst.ReadNewickFile(credible_rooted_nwk_path);
+  pp_r_inst.ReadNewickFile(pp_rooted_nwk_path);
+
+  auto all_trees = all_trees_gp_inst.GenerateCompleteRootedTreeCollection();
+  auto cred_trees = cred_r_inst.tree_collection_;
+  auto pp_trees = pp_r_inst.tree_collection_;
+  auto indexer = all_trees_gp_inst.GetDAG().BuildGPCSPIndexer();
+  std::vector<RootedIndexerRepresentation> all_representations;
+  std::vector<RootedIndexerRepresentation> cred_representations;
+  std::vector<RootedIndexerRepresentation> pp_representations;
+  LoadIndexerRepresentations(all_representations, all_trees, indexer);
+  LoadIndexerRepresentations(cred_representations, cred_trees, indexer);
+  LoadIndexerRepresentations(pp_representations, pp_trees, indexer);
+
+  UnrootedSBNInstance ur_inst("charlie");
+  ur_inst.ReadNewickFile(rooted_nwk_path);
+  ur_inst.ReadFastaFile(fasta_path);
+  PhyloModelSpecification simple_specification{"JC69", "constant", "strict"};
+  ur_inst.PrepareForPhyloLikelihood(simple_specification, thread_count, {}, true,
+                                    all_trees.TreeCount());
+  const auto log_likelihoods = ur_inst.UnrootedLogLikelihoods(all_trees);
+
+  WriteTreesToFile(out_path, all_representations, log_likelihoods);
+  WriteTreesToFile(credible_out_path, cred_representations);
+  WriteTreesToFile(pp_out_path, pp_representations);
+}
+
+void LoadIndexerRepresentations(
```

Comment:
It seems like we could rewrite this to just return the indexer representation, which is stylistically preferable to mutating it in place.

Reply:
I must have been going back and forth about pass by value and pass by reference with old school C++.

I've pushed a new version.